End to end selenium tests (Stripe and Paypal)

jranks123 opened the pull request:
This PR adds selenium end to end tests for the Stripe and Paypal checkout flows. 

The tests can be run locally with the command `sbt test-only *StripeCheckoutSpec*

jranks123 committed:
set up selinium test

jranks123 committed:
pushing up for Sam

desbo committed:
Use Driver object and add some page helpers

jranks123 committed:
selenium tests implemented

desbo commented on test/acceptance/StripeCheckoutSpec.scala:
```diff
@@ -0,0 +1,67 @@
+  val postPayment = pages.PostPayment
+
+
+    "The OneBrowserPerTest trait" must {
```
indentation is a little wild in this file

michaelwmcnamara commented:
just tried locally - you need a local instance of contributions-frontend up and running.
Once this is up and running, if you run the command "sbt test-only StripeCheckoutSpec.scala":
All tests seem to run and pass, however there is an error message at the end of the run:

"[error] No valid parser available.
[error] StripeCheckoutSpec.scala"

Commit:
copying and adapting the backgroud setup from membership-frontend so that we can use the same travis and saucelabs setup

michaelwmcnamara commented:
Added background setup copied from membership-frontend. This will allow us to use the same method as they do to run our tests - specifically a hook from prout to Travis, which then runs the tests on saucelabs.

Commit:
fix paypal sandbox email credentials

desbo committed:
Add test username cookie

This allows the Selenium tests to switch the gateways into test mode by
providing a valid test identity username in the _test_username cookie

desbo committed:
Tidy up and remove files we don't need

desbo committed:
Remove package-lock.json

desbo committed:
Add .travis.yml

Commit:
Remove unused config value

Commit:
Update prout.json to run acceptance test in Travis

Commit:
Only run normal unit tests in .travis.yml

Prout will trigger the full acceptance test suite

Commit:
Remove unused config value

desbo committed:
Fix broken Travis build

desbo committed:
Tweaks to get it working

desbo committed:
Better assertion in PayPal step

desbo committed:
Add note to README about running selenium tests

joelochlann reviewed (approved):
Looks good to me
